book/src/api-reference/jsonrpc-api.md

```diff
@@ -149,6 +150,34 @@ curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0", "id":1, "
 {"jsonrpc":"2.0","result":0,"id":1}
 ```
 
+### getBlockConfidence
+
+Returns confidence for particular block
+
+#### Parameters:
+
+* `u64` - fork, identified by slot id
```

Comment:
`block, identified by block id`?

Reply:
That's not quite right. Block id is the last entry id (hash) in the block, and this is just returning the Slot.

Reply:
ah, k.  So if we have a virtual slot then we'd end up with a `null` response?

Reply:
What's a virtual slot?

Reply:
A slot that produced no block.

Reply:
Gotcha! Yes, that should be correct, because such a slot would not be cached in the ForkConfidenceCache